src/backend/distributed/connection/connection_configuration.c

```diff
@@ -341,6 +341,31 @@ GetConnParam(const char *keyword)
 }
 
 
+/*
```

Comment:
I'm trying not to change existing routines much, but I'm not that keen on this routine, another route could be

FindOrCreateConnParamsEntry(....,app-name)->GetConnParams->(...,app-name) and set it in the routine.

and completely avoid this new function.

Reply:
We are also changing the similar things on https://github.com/citusdata/citus/pull/5601/files

I'm trying to think of a way what we can rely on for both PRs. I'll try to reply back today with a better understanding of both PRs

Reply:
Thanks

Reply:
I think the slight disadvantage of this approach is that for any other such internal connections, we might end up having different flags (e.g., `run_command_on_workers` flag etc.).

Also, rebalancer always used with `FORCE_NEW_CONNECTION` which makes sure that we always have a new connection hence use the new application_name. Citus normally caches connection, and unless `FORCE_NEW_CONNECTION` is used, the existing cached connection is used. 

So, how can we make this more generic? The naive idea I had was noted [here](https://github.com/citusdata/citus/pull/5601/files#r793463283), which I do not prefer much as well. Because, that is also not going to work in a generic way considering the cached connections.

The more generic way (considering the cached connections) might be to send `SET application_name TO xxx` over the connection, and once we are done, `RESET application_name` such that the globalPID we assign stays on the connection.

So, something like:

```SQL
void
ExecuteRebalancerCommandInSeparateTransaction(char *command)
{
	int connectionFlag = FORCE_NEW_CONNECTION;


	MultiConnection *connection = GetNodeConnection(connectionFlag, LocalHostName,
													PostPortNumber);
	ExecuteCriticalRemoteCommand(connection,"SET application_name TO rebalancer");

	ExecuteCriticalRemoteCommand(connection, command);
	ExecuteCriticalRemoteCommand(connection,"RESET application_name");

	CloseConnection(connection);
```

Though, we still need to find a way to clear the application name if  `command` fails.

Do you have any further thoughts/ideas? It is Friday afternoon, so this might not be the a cool idea :)

Reply:
I thought about global variable approach, but resetting is a problem depending on which code path you set the value.

Reply:
I've added other approach too, which makes this routine redundant

Reply:
Chnaged it to the SET model